From 56c07a487d2b836968469a7acff3515211c36c73 Mon Sep 17 00:00:00 2001
From: Antonio Vázquez <antoniovazquezblanco@gmail.com>
Date: Tue, 10 Mar 2026 19:02:21 +0100
Subject: [PATCH] dmi: Avoid redefinition of _POSIX_C_SOURCE

This fixes compilation errors due to redefinitions of the
_POSIX_C_SOURCE macro in dmi.c when in Linux and compiling
using gcc 15.1.1.

Change-Id: I8c70fc8cad57e29719d68032e3d2ecba7e9f5932
Signed-off-by: Antonio Vázquez <antoniovazquezblanco@gmail.com>
Reviewed-on: https://review.coreboot.org/c/flashrom/+/91643
Reviewed-by: Anastasia Klimchuk <aklm@chromium.org>
Tested-by: build bot (Jenkins) <no-reply@coreboot.org>
Reviewed-by: Abdelkader Boudih <coreboot@seuros.com>
---

diff --git a/dmi.c b/dmi.c
index d0c68de..76eb05d 100644
--- a/dmi.c
+++ b/dmi.c
@@ -12,8 +12,10 @@
 #if (__GLIBC__ == 2 && __GLIBC_MINOR__ < 10) || __GLIBC__ < 2
 #define _GNU_SOURCE
 #else
+#if !defined(_POSIX_C_SOURCE)
 #define _POSIX_C_SOURCE 200809L
 #endif
+#endif
 
 #include "dmi.h"
 
